main.tf

```diff
@@ -219,22 +190,18 @@ resource "aws_cloudfront_distribution" "cloudfront_distribution" {
   }
 
   dynamic "logging_config" {
-    for_each = [for i in var.dynamic_logging_config : {
-      bucket          = i.bucket
-      include_cookies = i.include_cookies
-      prefix          = i.prefix
-    }]
+    for_each = var.dynamic_logging_config[*]
 
     content {
       bucket          = logging_config.value.bucket
-      include_cookies = logging_config.value.include_cookies
-      prefix          = logging_config.value.prefix
+      include_cookies = lookup(logging_config.value, "include_cookies", null)
+      prefix          = lookup(logging_config.value, "prefix", null)
     }
   }
 
-  tags = {
-    Name = var.tag_name
-  }
+  tags = merge(
+    var.additional_tags,
+  { Name = var.tag_name })
```

Comment:
I like this idea!

Reply:
Best would have been to have only 1 variable for tags, but to maintain the backward compability, i've left this, usually i prefer messing with the tags in local variables on the implementation

Reply:
agreed, appreciate you leaving backwards compatability